### PR TITLE
Resolves GravityGeneratorVisualizer is Obsolete

### PR DESCRIPTION
**Pull request by TemporalOroboros**
## About the PR

Splits the GravityGeneratorVisualizer into a component/system pair.

### TODO:
- [ ] Testing

**Media**

- [ ] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

### CODE REVIEW

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
Just use the gravitygeneratorcomponent

**Comment by TemporalOroboros** [comment]
> Just use the gravitygeneratorcomponent

Do you want me to make a client version of SharedGravityGeneratorComponent or move the sprite map to shared (and therefor the server)?

**Comment by metalgearsloth** [comment]
The status should be on shared; using a separate comp state for it would also potentially mean removing appearancecomponent.

File: Content.Client/Gravity/GravitySystem.cs [hunk 1]
```diff
@@ -1,12 +1,79 @@
...
+    /// Ensures that gravity generators have the sprite layers necessary to be rendered correctly.
+    /// </summary>
+    private void OnComponentStartup(EntityUid uid, SharedGravityGeneratorComponent comp, ComponentStartup args)
+    {
+        if(!TryComp<SpriteComponent>(uid, out var sprite))
+            return;
+
+        sprite.LayerMapReserveBlank(GravityGeneratorVisualLayers.Base);
+        sprite.LayerMapReserveBlank(GravityGeneratorVisualLayers.Core);
+    }
```
Comment (AJCM-git) [inline Content.Client/Gravity/GravitySystem.cs:27]: i dont think this is necessary

File: Content.Shared/Gravity/SharedGravityGeneratorComponent.cs [hunk 2]
```diff
@@ -7,6 +7,37 @@ namespace Content.Shared.Gravity
...
+        /// <summary>
+        /// The sprite used by the core of the gravity generator when the gravity generator is activating.
+        /// </summary>
+        [DataField("coreActivatingState")]
+        public string CoreActivatingState = "activating";
+        
+        /// <summary>
+        /// The sprite used by the core of the gravity generator when the gravity generator is active.
+        /// </summary>
+        [DataField("coreActivatedState")]
```
Comment (AJCM-git) [inline Content.Shared/Gravity/SharedGravityGeneratorComponent.cs:38]: VV

**Review by AJCM-git — CHANGES_REQUESTED** [review]

File: Content.Client/Gravity/GravitySystem.cs (hunk 1 above)
Comment (TemporalOroboros) [reply Content.Client/Gravity/GravitySystem.cs:27]: Turns out it wasn't.

File: Content.Shared/Gravity/SharedGravityGeneratorComponent.cs (hunk 2 above)
Comment (TemporalOroboros) [reply Content.Shared/Gravity/SharedGravityGeneratorComponent.cs:38]: Done.

**Comment by TemporalOroboros** [comment]
@DrSmugleaf 
Pushing commits isn't reopening the PR. Please reopen it at your convenience.

**Review by metalgearsloth — APPROVED** [review]

**Pull request merged**